src/cudamatrix/cu-vector.cc

```diff
@@ -328,7 +328,7 @@ MatrixIndexT CuVectorBase<Real>::ApplyFloor(Real floor_val) {
     cuda_vec_apply_floor(dimGrid, dimBlock, data_, floor_val, count_vec.Data(), dim_);
     CU_SAFE_CALL(cudaGetLastError());
     num_floored = count_vec.Sum();
-    CuDevice::Instantiate().AccuProfile("CuVectorBase::ApplyFloor", tim.Elapsed());
+    CuDevice::Instantiate().AccuProfile(__func__, tim.Elapsed());
```

Comment:
Daniel's right, that's not what we want because there is a similarly named
function in class CuMatrix that we want to distinguish.


On Mon, Jul 11, 2016 at 3:38 PM, Daniel Galvez <notifications@github.com>
wrote:

> In src/cudamatrix/cu-vector.cc
> <https://github.com/kaldi-asr/kaldi/pull/803#discussion_r70350264>:
>
> > @@ -328,7 +328,7 @@ MatrixIndexT CuVectorBase<Real>::ApplyFloor(Real floor_val) {
> >      cuda_vec_apply_floor(dimGrid, dimBlock, data_, floor_val, count_vec.Data(), dim_);
> >      CU_SAFE_CALL(cudaGetLastError());
> >      num_floored = count_vec.Sum();
> > -    CuDevice::Instantiate().AccuProfile("CuVectorBase::ApplyFloor", tim.Elapsed());
> > +    CuDevice::Instantiate().AccuProfile(__func__, tim.Elapsed());
>
> To be specific, __func__ outputs only "ApplyFloor" here. It doesn't
> include the class name. It's that C legacy.
> … <#m_-7613624424334759022_>
> On Sun, May 22, 2016 at 1:39 PM, Daniel Povey ***@***.***> wrote: In
> src/cudamatrix/cu-vector.cc <#803 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/803#discussion_r64155921>>: > @@
> -328,7 +328,7 @@ MatrixIndexT CuVectorBase<Real>::ApplyFloor(Real
> floor_val) { > cuda_vec_apply_floor(dimGrid, dimBlock, data_, floor_val,
> count_vec.Data(), dim_); > CU_SAFE_CALL(cudaGetLastError()); > num_floored
> = count_vec.Sum(); > -
> CuDevice::Instantiate().AccuProfile("CuVectorBase::ApplyFloor",
> tim.Elapsed()); > + CuDevice::Instantiate().AccuProfile(__func__,
> tim.Elapsed()); Yiming, the reason we sometimes write e.g.
> CuVectorBase::ApplyFloor instead of *func*, is that when the same function
> exists in multiple classes (e.g. also in CuMatrixBase), we otherwise
> wouldn't know which one was taking time. — You are receiving this because
> you are subscribed to this thread. Reply to this email directly or view it
> on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/803/files/5020094e0f6a6a4a407c3924367bc608e83a7323#r64155921>
>
> -- Daniel Galvez
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/803/files/5020094e0f6a6a4a407c3924367bc608e83a7323#r70350264>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADJVuyOV5Q_GS1v6vUNCvBSL3VmTXfq-ks5qUsYAgaJpZM4IkDwb>
> .
>